Source/messaging/MessageClient.cpp

```diff
@@ -176,21 +179,46 @@ namespace Messaging {
                     size = sizeof(_readBuffer);
                 }
 
-                auto length = information.Deserialize(_readBuffer, size);
+                if (metadata.Deserialize(_readBuffer, size) != 0) {
+
+                    if (metadata.Type() == Core::Messaging::Metadata::type::TRACING) {
+                        auto length = trace.Deserialize(_readBuffer, size);
```

Comment:
What is the type of the length variable?

Reply:
It used to be auto before my changes, but it is simply uint16_t, so there is no point to keep it as auto I guess. Should I just change it to uint16_t?

Reply:
yes please